tdrs-backend/scripts/db_backup.py

```diff
@@ -0,0 +1,228 @@
+    s3.download_file(bucket, object_name, file_name)
+
+
+def list_s3_files(bucket,
+                  region):
+    """List the files in s3 bucket."""
+    """
+    :param bucket:
+    :param region:
+    """
+    session = boto3.Session(aws_access_key_id=S3_ACCESS_KEY_ID,
+                            aws_secret_access_key=S3_SECRET_ACCESS_KEY,
+                            region_name=region)
+    s3 = session.resource('s3')
+    my_bucket = s3.Bucket(bucket)
+    return list(_ for _ in my_bucket.objects.all())
+
+
+def get_database_credentials(database_uri):
+    """Export database credentials from database URI."""
+    """
+    :param database_uri: Database URI as postgresql://$<USERNAME>:$<PASSWORD>@$<HOST>:$<PORT>/$<NAME>
+    """
+    database_uri = database_uri[database_uri.find('//')+2:]
+    username = database_uri[:database_uri.find(':')]
+    database_uri = database_uri[database_uri.find(':')+1:]
+    password = database_uri[:database_uri.find('@')]
+    database_uri = database_uri[database_uri.find('@') + 1:]
+    host = database_uri[:database_uri.find(':')]
+    database_uri = database_uri[database_uri.find(':') + 1:]
+    port = database_uri[:database_uri.find('/')]
+    database_uri = database_uri[database_uri.find('/') + 1:]
+    database_name = database_uri
+    return [username, password, host, port, database_name]
+
+
+def handle_args(argv):
+    """Handle commandline args."""
+    arg_file = "/tmp/backup.pg"
+    arg_database = DATABASE_URI
+    arg_to_restore = False
+    arg_to_backup = False
+
+    try:
+        opts, args = getopt.getopt(argv, "hbr:f:d:", ["help", "backup", "restore", "file", "database", ])
+        for opt, arg in opts:
+            if "backup" in opt or "-b" in opt:
+                arg_to_backup = True
+            elif "restore" in opt or "-r" in opt:
+                arg_to_restore = True
+            if "file" in opt or "-f" in opt:
+                arg_file = arg if arg[0] == "/" else "/tmp/" + arg
+            if "database" in opt or "-d" in opt:
+                arg_database = arg
+
+    except Exception as e:
+        print(e)
+        sys.exit(1)
+
+    if arg_to_backup:
+        # back up database
+        backup_database(file_name=arg_file,
+                        postgres_client=POSTGRES_CLIENT,
+                        database_uri=arg_database)
+
+        # upload backup file
+        upload_file(file_name=arg_file,
+                    bucket=S3_BUCKET,
+                    region=S3_REGION,
+                    object_name="/backup"+arg_file)
+
+    elif arg_to_restore:
+        # download file from s3
+        download_file(bucket=S3_BUCKET,
+                      file_name=arg_file,
+                      region=S3_REGION,
+                      object_name="/backup"+arg_file)
+
+        # restore database
+        restore_database(file_name=arg_file,
+                         postgres_client=POSTGRES_CLIENT,
+                         database_uri=arg_database)
```

Comment:
Same as with uploading, can we delete the dump file upon a successful restoration?